src/storage/checkpoint_manager.cpp

```diff
@@ -0,0 +1,77 @@
+#include "storage/checkpoint_manager.h"
+#include <vector>
+
+#define NUM_RESERVED_COLUMNS 1u
+
+namespace terrier::storage {
+void CheckpointManager::Checkpoint(const SqlTable &table, const BlockLayout &layout) {
+  std::vector<storage::col_id_t> all_col(layout.NumColumns() - NUM_RESERVED_COLUMNS);
+  // Add all of the column ids from the layout to the projection list
+  // 0 is version vector so we skip it
+  for (uint16_t col = NUM_RESERVED_COLUMNS; col < layout.NumColumns(); col++) {
+    all_col[col - NUM_RESERVED_COLUMNS] = storage::col_id_t(col);
+  }
+
+  // TODO(Mengyang): should calculate number of tuple to fit in buffer_size_
+  // now I only use a magic number here
+  uint32_t max_tuples = 10;
+
+  ProjectedColumnsInitializer column_initializer(layout, all_col, max_tuples);
+  auto *scan_buffer = common::AllocationUtil::AllocateAligned(column_initializer.ProjectedColumnsSize());
+  ProjectedColumns *columns = column_initializer.Initialize(scan_buffer);
+
+  ProjectedRowInitializer row_initializer = ProjectedRowInitializer::CreateProjectedRowInitializer(layout, all_col);
+  auto *redo_buffer = common::AllocationUtil::AllocateAligned(row_initializer.ProjectedRowSize());
+  ProjectedRow *row_buffer = row_initializer.InitializeRow(redo_buffer);
+
+  auto it = table.begin();
+  auto end = table.end();
+  while (it != end) {
+    table.Scan(txn_, &it, columns);
```

Comment:
Using Select here instead of Scan can let you skip the entire part of trying to create a ProjectedRow in serialize tuple, you will just get a ProjectedRow back